src/Functions/createLink.js

```diff
@@ -0,0 +1,61 @@
+import { pick } from 'lodash';
+import HashId from 'hashids/cjs';
+import { info } from 'heroku-logger';
+import * as Express from 'express';
+
+import config from '../../config';
+import Link from '../Models/Link';
+import { randomChar } from '../helpers';
+
+/**
+ * Crete a new shortlink key.
+ *
+ * @return {String}
+ */
+async function generateKey() {
+  // In order to avoid a single "hot" key with our counter, we split
+  // the table into 3,844 partitions (e.g. AA, AB, AC...). Each partition
+  // keeps its own auto-incremented counter for making shortlinks, below:
+  const partition = await Link.update(
+    { key: `${randomChar()}${randomChar()}` },
+    { $ADD: { counter: 1 } }
```

Comment:
Is this some specific property key or something? `$ADD`. Just unfamiliar.

Reply:
Yep! This is a special property on the [`Model.update`](https://dynamoosejs.com/guide/Model#modelupdatekey-updateobj-settings-callback) payload that maps to this [DynamoDB query](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html#Expressions.UpdateExpressions.SET.IncrementAndDecrement). It allows us to increment this counter without a multi-step read & write transaction.